src/widgets/transactions-widget/transaction-row-widget.tsx

```diff
@@ -0,0 +1,142 @@
+import React, {useCallback, useMemo} from 'react';
+
+import {formatDistance} from 'date-fns';
+import {TouchableWithoutFeedback, View} from 'react-native';
+
+import {Color} from '@app/colors';
+import {DataContent, Icon, IconsName, Text} from '@app/components/ui';
+import {createTheme} from '@app/helpers';
+import {cleanNumber} from '@app/helpers/clean-number';
+import {I18N} from '@app/i18n';
+import {Transaction} from '@app/models/transaction';
+import {Wallet} from '@app/models/wallet';
+import {CURRENCY_NAME} from '@app/variables/common';
+
+export type Props = {
+  item: Transaction;
+  onPress: (hash: string) => void;
+};
+
+type IMapItem = {
+  iconName: IconsName;
+  title: I18N;
+  sumTextColor: Color;
+  sign: string;
+};
+
+const DisplayMap: {[key: string]: IMapItem} = {
+  send: {
+    iconName: IconsName.arrow_send,
+    title: I18N.transactionSendTitle,
+    sumTextColor: Color.textRed1,
+    sign: '-',
+  },
+  receive: {
+    iconName: IconsName.arrow_receive,
+    title: I18N.transactionReceiveTitle,
+    sumTextColor: Color.textGreen1,
+    sign: '+',
+  },
+};
+
+export const TransactionRowWidget = ({item, onPress}: Props) => {
+  const wallets = useMemo(() => Wallet.addressList(), []);
```

Comment:
move Wallet.addressList in index.ts, where you get transactions. All queries to database should be on same page (in current case, in root)

Reply:
props drilling anti-pattern, but i'll do it